lib/ws-browser.js

```diff
@@ -36,6 +49,29 @@ class Transport extends EventEmitter {
     return this.socket;
   }
 
+  _sendBufferedMessages() {
+    let bufferedAmount = this.socket.bufferedAmount;
+    let toSend = '';
+    while (this._sendBuffer.length !== 0) {
+      const message = this._sendBuffer[0];
+      if (message.length + bufferedAmount <= BUFFER_SIZE_LIMIT) {
+        toSend += message;
+        bufferedAmount += message.length;
+        this._sendBuffer.shift();
+      } else {
+        const chunkLength = BUFFER_SIZE_LIMIT - bufferedAmount;
+        if (chunkLength !== 0) {
```

Comment:
I understand that this algorithm is made so that chunkLength is always >= 0 though I'd still prefer a >0 check here.

Reply:
If the `bufferedAmount` value is bigger than `BUFFER_SIZE_LIMIT` value, it must be considered an error, not a normal situation. Considering this, I think it's better to add the check for `bufferedAmount > BUFFER_SIZE_LIMIT` (or `chunkLenght < 0`) that will throw in such a situation.

Reply:
Well, I don't see how's that situation possible so I'm not sure if we should check for that.
All I wanted is for this check to be a bit more clear when you read the code, meaning - if chunkLen is bigger than 0 then we have something to write.

Reply:
This situation is possible if someone using the package will access the WebSocket connection directly and just call the `send()` method on it.